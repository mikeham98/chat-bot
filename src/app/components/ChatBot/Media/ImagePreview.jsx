import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../../../themes/components/media.scss';

const ImagePreview = ({src}) => {
    return (
        <img className={styles.mediaImagePreviewImage} src={src}/>
    );
};

ImagePreview.propTypes = {
    src: PropTypes.string.isRequired
};

export default React.memo(ImagePreview);