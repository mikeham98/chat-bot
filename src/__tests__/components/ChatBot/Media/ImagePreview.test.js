import React from 'react';
import {shallow} from 'enzyme';
import ImagePreview from '../../../../app/components/ChatBot/Media/ImagePreview';

describe('ImagePreview', () => {

    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<ImagePreview {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            src: 'http://www.image.co.uk/img1'
        };
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return an img tag with the src set to \'http://www.image.co.uk/img1\'', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
});