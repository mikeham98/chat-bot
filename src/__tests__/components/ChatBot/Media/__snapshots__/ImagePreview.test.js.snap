// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreview render should return an img tag with the src set to 'http://www.image.co.uk/img1' 1`] = `
<img
  className="mediaImagePreviewImage"
  src="http://www.image.co.uk/img1"
/>
`;
